src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentsDetails/payment.js

```diff
@@ -0,0 +1,54 @@
+const Logger = require('dw/system/Logger');
+const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
+
+function getFinalResponse(paymentsDetailsResponse) {
+  return {
+    isFinal: true,
+    isSuccessful: paymentsDetailsResponse.resultCode === 'Authorised',
+    ...(['Refused', 'Error'].indexOf(paymentsDetailsResponse.resultCode) !==
+      -1 && { refusalReason: paymentsDetailsResponse.refusalReason }),
+  };
+}
+
+function handlePaymentsDetails(stateData) {
+  const paymentsDetailsResponse = adyenCheckout.doPaymentDetailsCall(stateData);
+
+  if (
+    ['Authorised', 'Refused', 'Error', 'Cancelled'].indexOf(
+      paymentsDetailsResponse.resultCode,
+    ) !== -1
+  ) {
+    return getFinalResponse(paymentsDetailsResponse);
+  }
+
+  if (
+    [
```

Comment:
Perhaps we can extract this logic to a ResponseHandler, so we can re-use it when implementing this flow in the checkout. Let's discuss this offline